undistort_test.py

```diff
@@ -0,0 +1,86 @@
+import open3d as o3d
+import argparse
+import numpy as np
+import mrob
+import math
+
+from pathlib import Path
+from scipy.spatial.transform import Rotation as R
+from collections import defaultdict
+
+def get_intermediate_transform(T1, T2, t):
+    dxi = mrob.geometry.SE3(T2 @ np.linalg.inv(T1)).Ln()
+    return mrob.geometry.SE3(t * dxi).T() @ T1
+
+def undistort_segment(pcd, theta_min, theta_max, color, T):
+    # Convert the point cloud points to a numpy array
+    points = np.asarray(pcd.points)
+    thetas = np.asarray([math.atan2(p[1],p[0]) for p in points])
+
+    # Create a mask for points that fall within the ROI
+    mask_roi = (thetas < theta_max) & (thetas >= theta_min)
+    trues = mask_roi.sum()
+
+    new_pcd = o3d.geometry.PointCloud()
+    if trues > 0:
+        new_colors = np.full((trues, 3), color)
+        new_points = points[mask_roi]
+
+        new_pcd.points = o3d.utility.Vector3dVector(new_points)
+        new_pcd.colors = o3d.utility.Vector3dVector(new_colors)
+
+    return new_pcd.transform(T)
+
+def undistort_cloud(open3d_cloud, color, N, T1, T2):
+    N += 1 # starting separations from 0
+    undistort_result = []
+    prev_sector = -np.pi
```

Comment:
```suggestion
    previous_sector = -np.pi
```

Let's use entire word in variables naming